addons/stock/wizard/stock_return_picking.py

```diff
@@ -133,10 +133,11 @@ def _create_returns(self, cr, uid, ids, context=None):
             new_qty = data_get.quantity
             if new_qty:
                 returned_lines += 1
+                values = move_obj.onchange_quantity(cr, uid, [], data_get.product_id.id, new_qty, move.product_uom.id, move.product_uos.id)
                 move_obj.copy(cr, uid, move.id, {
                     'product_id': data_get.product_id.id,
                     'product_uom_qty': new_qty,
-                    'product_uos_qty': uom_obj._compute_qty(cr, uid, move.product_uom.id, new_qty, move.product_uos.id),
+                    'product_uos_qty': values['value']['product_uos_qty'],
```

Comment:
What do you think about: 'product_uos_qty': new_qty / move.product_uom_qty * move.product_uos_qty,  I think it is also done that way when partially transferring something.